keycloak/sms-provider/src/main/java/org/sunbird/keycloak/KeycloakSmsAuthenticatorUtil.java

```diff
@@ -84,20 +88,50 @@ public static String setDefaultCountryCodeIfZero(String mobileNumber) {
     }
 
     static boolean sendSmsCode(String mobileNumber, String code, AuthenticatorConfigModel config) {
-        // Send an SMS
-        KeycloakSmsAuthenticatorUtil.logger.debug("Sending " + code + "  to mobileNumber " + mobileNumber);
+        String smsText = createMessage(code, mobileNumber, config);
+        logger.debug("KeycloakSmsAuthenticatorUtil@sendSmsCode : smsText - " + smsText);
 
-        String smsUsr = getConfigString(config, KeycloakSmsAuthenticatorConstants.CONF_PRP_SMS_CLIENTTOKEN);
-        String smsPwd = getConfigString(config, KeycloakSmsAuthenticatorConstants.CONF_PRP_SMS_CLIENTSECRET);
+//        Boolean amazonSmsProviderStatus = msgViaAmazonSns(mobileNumber, smsText);
+//        if (amazonSmsProviderStatus != null) return amazonSmsProviderStatus;
 
-        String smsText = createMessage(code, mobileNumber, config);
-        try {
-            PublishResult send_result = new SnsNotificationService().send(setDefaultCountryCodeIfZero(mobileNumber), smsText, smsUsr, smsPwd);
-            return true;
-        } catch (Exception e) {
-            //Just like pokemon
-            return false;
+        Boolean msg91SmsProviderStatus = sendViaMsg91(mobileNumber, smsText);
```

Comment:
This method looks incorrect. Here's what I suggest
- create a json config that has the name of the "currently active" provider
- read this config and initialise the appropriate factory
- invoke the send method on the factory

that way you avoid provider specific code here. This also ensures that more SMS providers can be easily added

Reply:
@coolbung - This is actually just for quick testing I had kept it. Ideally what should happen is, during the configuration of SMS provider from UI, admin will select what is the SMS provider and based on that respective provider send method will be called.